symengine/llvm_double.h

```diff
@@ -41,9 +69,14 @@ class LLVMDoubleVisitor : public BaseVisitor<LLVMDoubleVisitor>
 
 public:
     llvm::Value *apply(const Basic &b);
-    void init(const vec_basic &x, const Basic &b, bool cse = false);
+    void init(const vec_basic &x, const Basic &b,
+              const bool symbolic_cse = false,
+              const std::vector<llvm::Pass *> &passes = {});
     void init(const vec_basic &inputs, const vec_basic &outputs,
-              bool cse = false);
+              const bool symbolic_cse = false,
+              const std::vector<llvm::Pass *> &passes = {});
```

Comment:
Instead of this, how about 2 signatures like,
```C++
void init(const vec_basic &, const vec_basic &, bool symbolic_cse = false, int optlevel=2);
void init(const vec_basic &, const vec_basic &, bool symbolic_cse = false, const std::vector<llvm::Pass *> &);
```

Reply:
sounds reasonable (is it the first one that the compiler will choose for `init(inp, out)`?)

Reply:
yes. edited the 2nd signature.

Reply:
I don't think 2 different sets of default arguments only differing in types are allowed? (I get `error: call of overloaded ‘init(<brace-enclosed initializer list>, const SymEngine::Basic&)’ is ambiguous`)

Reply:
never mind: I forgot to remove defaults for the vector.